zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -160,6 +160,7 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         // no_callee_code is true when the account exists and has empty
         // code hash, or when the account doesn't exist (which we encode with
         // code_hash = 0).
+        // Why do we have summation here if any of the terms is enough?
         let no_callee_code = is_empty_code_hash.expr() + callee_not_exists.expr();
```

Comment:
Here we have:
- `is_empty_code_hash: code_hash == EMPTY_CODE_HASH`
- `callee_not_exists: code_hash == 0`

And we want
- `no_callee_code: is_empty_code_hash OR callee_not_exists`

But since `is_empty_code_hash` and `callee_not_exists` can't be true at the same time, doing an addition is the same as doing an OR (and it's a simpler constraint).

This is the truth table for `no_callee_code`

| no_callee_code | is_empty_code_hash | callee_not_exists |
| --- | --- | --- |
| False (0 = 0 + 0) | False (0) | False (0) |
| True (1 = 0 + 1) | False (0) | True (1) |
| True (1 = 1 + 0) | True (1) | False (0) |